aptos-move/move-examples/token_objects/ambassador/client/ambassador.ts

```diff
@@ -0,0 +1,144 @@
+    for (const ownership of tokenOwnership.current_token_ownerships_v2) {
+        if (ownership.current_token_data.token_name === tokenName) {
+            return new HexString(ownership.current_token_data.token_data_id);
+        }
+    }
+    console.log(`Token ${tokenName} not found`);
+    process.exit(1);
+}
+
+async function waitForEnter() {
+    return new Promise<void>((resolve, reject) => {
+        const rl = require("readline").createInterface({
+            input: process.stdin,
+            output: process.stdout
+        });
+
+        rl.question('Please press the Enter key to proceed ...\n', () => {
+            rl.close();
+            resolve();
+        });
+    });
+}
+
+class AmbassadorClient extends AptosClient {
+    constructor() {
+        super(NODE_URL);
+    }
+
+    async setAmbassadorLevel(creator: AptosAccount, token: HexString, new_ambassador_level: number | bigint): Promise<string> {
+        const rawTxn = await this.generateTransaction(creator.address(), {
+            function: `${creator.address()}::ambassador::set_ambassador_level`,
+            type_arguments: [],
+            arguments: [token.hex(), new_ambassador_level],
+        });
+
+        const bcsTxn = await this.signTransaction(creator, rawTxn);
+        const pendingTxn = await this.submitTransaction(bcsTxn);
+
+        return pendingTxn.hash;
+    }
+
+    async burn(creator: AptosAccount, token: HexString): Promise<string> {
+        const rawTxn = await this.generateTransaction(creator.address(), {
+            function: `${creator.address()}::ambassador::burn`,
+            type_arguments: [],
+            arguments: [token.hex()],
+        });
+
+        const bcsTxn = await this.signTransaction(creator, rawTxn);
+        const pendingTxn = await this.submitTransaction(bcsTxn);
+
+        return pendingTxn.hash;
+    }
+
+    async mintAmbassadorToken(creator: AptosAccount, description: string, name: string, uri: string, soul_bound_to: HexString): Promise<string> {
+        const rawTxn = await this.generateTransaction(creator.address(), {
+            function: `${creator.address()}::ambassador::mint_ambassador_token`,
+            type_arguments: [],
+            arguments: [description, name, uri, soul_bound_to.hex()],
+        });
+
+        const bcsTxn = await this.signTransaction(creator, rawTxn);
+        const pendingTxn = await this.submitTransaction(bcsTxn);
+
+        return pendingTxn.hash;
+    }
+
+    async ambassadorLevel(creator_addr: HexString, token_addr: HexString): Promise<bigint> {
+        const payload: Types.ViewRequest = {
+            function: `${creator_addr.hex()}::ambassador::ambassador_level`,
+            type_arguments: [],
+            arguments: [token_addr.hex()],
+        };
+
+        const result = await this.view(payload);
+        return BigInt(result[0].toString());
+    }
+}
+
+/** run our demo! */
+async function main(): Promise<void> {
+    const client = new AmbassadorClient();
```

Comment:
you dont need it, we [initialize](https://github.com/aptos-labs/aptos-core/pull/8200/files#diff-7207e18b0700e512a3b28cc6c31fa794b4ce636e9517d3ef48e5fd086ec6a0d8R7) a Provider class so can simply use that, i.e every `client.` can become `provider.`

Reply:
Done.